### DISCUSSION TITLE
Is there any documentation on how matching works?

### DISCUSSION THREAD

**Discussion by rea1shane** [Pagefind/pagefind#613] [Q&A]
I want to understand how it works. Like why is it matching here, but it shouldn't match.

<img width="1049" alt="image" src="https://github.com/CloudCannon/pagefind/assets/93363751/761b6be4-b6fb-48f0-ad40-5b29c75eede8">

**Comment by bglw** [comment]
No current documentation since it's sometimes a bit fuzzy, but in the above case:

- The pagefind site doesn't include `menu` anywhere, so there would be zero results
- Instead of returning nothing, Pagefind shortens the search word to just `men`
- That also has zero results, so it shortens again to `me`
- It finds the `me` portion of `rknight.me` due to the period splitting the words, and marks it as a result

There are a couple things to improve from Pagefind's side here:

1. Provide either default or configurable limits to this, so it matches "fuzzier" less
2. Show in the UI something like `Showing results for *me*` to make it clearer what Pagefind has done

**Reply by rea1shane** [reply]
Hope there will be documentation in the future explaining how it works and what I need to pay attention to when searching. And also hope that the improvements you mentioned can come soon.

Thank you for your work, it's a very nice project!

**Reply by rea1shane** [reply]
<img width="249" alt="截屏2024-05-14 00 47 45" src="https://github.com/CloudCannon/pagefind/assets/93363751/e2aaa14a-acec-4be8-bab8-bb2f11c15fa2">

~~Can we do like this now or later? When I search `gist`, result contains `gis` and `t`.~~

Oh, I'm sorry. I just realized that doesn't seem to be the same thing. What I described is more like https://pagefind.app/docs/multilingual/#specialized-languages. Looking forward to it coming soon.
